Sensors/SMCProcessor/SMCProcessor.cpp

```diff
@@ -56,9 +61,112 @@ void SMCProcessor::readRapl() {
 	}
 }
 
-void SMCProcessor::updateCounters() {
-	uint32_t cpu = cpu_number();
+IOReturn SMCProcessor::bindCurrentThreadToCpu(uint32_t cpu)
+{
+	// Obtain power management callbacks
+	pmCallBacks_t callbacks {};
+	pmKextRegister(PM_DISPATCH_VERSION, nullptr, &callbacks);
+	
+	if (!callbacks.LCPUtoProcessor) {
+		SYSLOG("scpu", "failed to obtain LCPUtoProcessor");
+		return KERN_FAILURE;
+	}
+	
+	if (!callbacks.ThreadBind) {
+		SYSLOG("scpu", "failed to obtain ThreadBind");
+		return KERN_FAILURE;
+	}
+	
+	auto preemptionLock = IOSimpleLockAlloc();
+	if (!preemptionLock) {
+		SYSLOG("scpu", "Preemption lock cannot be allocated");
+		return KERN_FAILURE;
+	}
+	
+	if (!IOSimpleLockTryLock(preemptionLock)) {
+		SYSLOG("scpu", "Preemption cannot be disabled before performing ThreadBind");
+		IOSimpleLockFree(preemptionLock);
+		return KERN_FAILURE;
+	}
+	
+	bool success = true;
+	auto enable = ml_set_interrupts_enabled(FALSE);
+	
+	while (1)
+	{
```

Comment:
This code looks really weird. Probably in Dell it was much longer, but here it makes sense to make it a simple if.